iodata/formats/mwfn.py

```diff
@@ -0,0 +1,367 @@
+    keywords = ["Shell types", "Shell centers", "Shell contraction"]
+    shell_types, shell_centers, prim_per_shell = _load_helper_shells(lit, nshell, keywords)
+    # HORTON indices start at 0 because Pythons do.
+    shell_centers -= 1
+    assert wfntype < 5
+    assert num_atoms > 0
+    assert min(atnums) >= 0
+    assert len(shell_types) == nshell
+    assert len(shell_centers) == nshell
+    assert len(prim_per_shell) == nshell
+    exponent = _load_helper_prims(lit, nprimshell)
+    coeffs = _load_helper_prims(lit, nprimshell)
+    # number of MO's should equal number of independent basis functions. MWFN inc. virtual orbitals.
+    num_coeffs = nindbasis
+    if wfntype in [0, 2, 3]:
+        # restricted wave function
+        num_mo = nindbasis
+    elif wfntype in [1, 4]:
+        # unrestricted wavefunction
+        num_mo = 2 * nindbasis
+
+    mo_numbers = np.empty(num_mo, int)
+    mo_type = np.empty(num_mo, int)
+    mo_occs = np.empty(num_mo, float)
+    mo_sym = np.empty(num_mo, str)
+    mo_energies = np.empty(num_mo, float)
+    mo_coeffs = np.empty([num_coeffs, num_mo], float)
+
+    for mo in range(num_mo):
+        mo_numbers[mo], mo_occs[mo], mo_energies[mo], mo_coeffs[:, mo], \
+            mo_type[mo], mo_sym[mo] = _load_helper_mo(lit, num_coeffs)
+
+    # TODO add density matrix and overlap
+
+    return {'title': title, 'energy': energy, 'wfntype': wfntype,
+            'nelec_a': nelec_a, 'nelec_b': nelec_b, 'charge': charge,
+            'atnums': atnums, 'atcoords': atcoords, 'atcorenums': atcorenums,
+            'nbasis': nbasis, 'nindbasis': nindbasis, 'nprims': nprim,
+            'nshells': nshell, 'nprimshells': nprimshell, 'full_virial_ratio': vt_ratio,
+            'shell_centers': shell_centers, 'shell_types': shell_types,
+            'prim_per_shell': prim_per_shell, 'exponents': exponent, 'coeffs': coeffs,
+            'mo_numbers': mo_numbers, 'mo_occs': mo_occs, 'mo_energies': mo_energies,
+            'mo_coeffs': mo_coeffs, 'mo_type': mo_type, 'mo_sym': mo_sym}
+
+
+def build_obasis(shell_map: np.ndarray, shell_types: np.ndarray,
+                 exponents: np.ndarray, prim_per_shell: np.ndarray,
+                 coeffs: np.ndarray,
+                 ) -> Tuple[MolecularBasis]:
+    """Based on the fchk modules basis building.
+
+    Parameters
+    -------------
+    shell_map:  np.ndarray (integer)
+        Index of what atom the shell is centered on. The mwfn file refers to this section
+        as `Shell centers`. Mwfn indices start at 1, this has been modified and starts
+        at 0 here. For water (O, H, H) with 6-31G, this would be an array like
+        [0, 0, 0, 0, 0, 1, 1, 2, 2]. , `O` in 6-31G has 5 shells and`H` has two shells.
+    shell_types: np.ndarray (integer)
+        Angular momentum of the shell. Indices start at 0 for 's' orbital, 1 for 'p' etc.
+        For 6-31G for a heavy atom this would be [0, 0, 1, 0, 1] corresponding
+         to [1s, 2s, 2p, 2s, 2p]
```

Comment:
This may break sphinx formatting:
```suggestion
        to [1s, 2s, 2p, 2s, 2p]
```